rotkehlchen/history/events.py

```diff
@@ -300,39 +303,55 @@ def get_history(
         history: List['AccountingEventMixin'] = []
         empty_or_error = ''
 
-        def populate_history_cb(
-                trades_history: List[Trade],
-                margin_history: List[MarginPosition],
-                result_asset_movements: List[AssetMovement],
-                exchange_specific_data: Any,
-        ) -> None:
-            """This callback will run for succesfull exchange history query
-
-            We don't include ledger actions here since we simply gather all of them at the end
-            """
-            history.extend(trades_history)
-            history.extend(margin_history)
-            history.extend(result_asset_movements)
-
-            if exchange_specific_data:
-                pass  # this used to be only for polo loans -- removed now. TODO: Think if needed
-
         def fail_history_cb(error_msg: str) -> None:
             """This callback will run for failure in exchange history query"""
             nonlocal empty_or_error
             empty_or_error += '\n' + error_msg
 
+        def new_step_cb(state_name: str) -> None:
+            """This callback will run for each new step in exchange history query"""
+            nonlocal step
+            step = self._increase_progress(step, total_steps)
+            self.processing_state_name = state_name
+
         for exchange in self.exchange_manager.iterate_exchanges():
             self.processing_state_name = f'Querying {exchange.name} exchange history'
-            exchange.query_history_with_callbacks(
+            exchange.query_history_with_fail_callback(
                 # We need to have history of exchanges since before the range
                 start_ts=Timestamp(0),
                 end_ts=end_ts,
-                success_callback=populate_history_cb,
                 fail_callback=fail_history_cb,
+                new_step_callback=new_step_cb,
```

Comment:
Hm ... you always pass it. Why did you make it optional and added all these other if checks in the function? Wtf? Just forget what I told you in that part then. Just pass both as non optional and remove those unneeded checks.

Reply:
ah fair enough. Did not see this since it was hidden in my review view as unchanged since last view.